packages/build-info/src/file-system.ts

```diff
@@ -179,11 +179,19 @@ export abstract class FileSystem {
   }
 
   /** Gracefully reads a file as JSON and parses it */
-  async readJSON<V = Record<string, unknown>>(path: string): Promise<Partial<V>> {
+  async readJSON<V = Record<string, unknown>>(path: string, options: { fail?: boolean } = {}): Promise<Partial<V>> {
+    if (options.fail === true) {
+      return JSON.parse(await this.readFile(path))
+    }
+
     try {
       return JSON.parse(await this.readFile(path))
     } catch (error) {
-      this.logger.error(`Could not parse JSON file ${path}\n${error}`)
+      let message = `Could not parse JSON file ${path}`
```

Comment:
nit: Feels a bit odd to be repeating the logic for reading and parsing the file just because of the `options.fail` branch. We could just re-throw inside the `catch`?

```suggestion
      if (options.fail === true) {
        throw error
      }

      let message = `Could not parse JSON file ${path}`
```

Reply:
yep good idea

Reply:
Addressed in https://github.com/netlify/build/pull/4977/commits/ac28711e57721593b35986c1b39480354cba73af